docs/graphkb/scripting/Annotate_a_VCF_with_GraphKB.ipynb

```diff
@@ -0,0 +1,1142 @@
+      },
+      "source": [
+        "Then we will iterate over the variant VCF records. From here we will need to pull out the HGVSp notation. Since this is output from SnpEff we can refer to their [documentation](https://pcingola.github.io/SnpEff/se_inputoutput/#eff-field-vcf-output-files) for how to parse this. According to their site this is the expected format of the `EFF` field which contains the HGVS notation. This field is pipe delimited inside parentheses.\n",
+        "\n",
+        "```text\n",
+        "EFF= Effect ( Effect_Impact | Functional_Class | Codon_Change | Amino_Acid_Change| Amino_Acid_Length | Gene_Name | Transcript_BioType | Gene_Coding | Transcript_ID | Exon_Rank  | Genotype_Number [ | ERRORS | WARNINGS ] )\n",
+        "```\n",
+        "\n",
+        "**Note**: This is a deprecated field. Newer versions use the `ANN` field. Parsing should be similar.\n",
+        "\n",
+        "We define the following functions below to help with parsing the INFO fields in the VCF"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "6IyqnAROXl_c"
+      },
+      "source": [
+        ""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "fXw09yCzv0FB"
+      },
+      "source": [
+        "import re \n",
+        "\n",
+        "\n",
+        "def parse_eff(eff):\n",
+        "  match = re.match(r'^[^(]+\\((.+)\\)$', eff)\n",
+        "  if match:\n",
+        "    [impact, functional_class, codon_change, aa_change, _, gene_name, transcript_biotype, gene_coding, transcript_id, _, genotype_allele ] = match.group(1).split('|')[:11]\n",
+        "\n",
+        "    changes = aa_change.split('/')\n",
+        "    protein_changes = [p for p in changes if p.startswith('p.')]\n",
+        "    cds_changes = [p for p in changes if p.startswith('c.')]\n",
+        "    \n",
+        "    return [\n",
+        "      impact, \n",
+        "      gene_name,\n",
+        "      transcript_biotype,\n",
+        "      gene_coding, \n",
+        "      transcript_id, \n",
+        "      protein_changes[0] if protein_changes else None, \n",
+        "      cds_changes[0] if cds_changes else None,\n",
+        "      genotype_allele\n",
+        "    ]\n",
+        "  raise NotImplementedError('cannot parse eff string. Does not match the expected pattern')\n"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nZLx2prpw3zP"
+      },
+      "source": [
+        "Now we want to convert the parsed content into a JSON/Dict to match the fields that we expect in IPR. If you are not uploading to IPR you can leave variants in their original format instead.\n",
+        "\n",
+        "**Note**: Some of the fields (support counts) require picking the samples. By default we have assumed the names are NORMAL and TUMOR. These may need to be customized for other input"
```

Comment:
maybe "some of the fields (e.g., support counts)"

Reply:
Its not an example field but more a grouping for fields so the e.g. might be confusing here